packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts

```diff
@@ -292,8 +307,14 @@ export class GmailTrigger implements INodeType {
 					{},
 					qs,
 				);
-
-				if (!simple) {
+				if (!includeDrafts) {
+					if (responseData[i].labelIds.includes('DRAFT')) {
+						responseData.splice(i, 1);
+						i--;
```

Comment:
```suggestion
						
```

Reply:
My reasoning was that the splicing shifts the next element to the current index and because we wanna check that element too we would have to reset the index.

Reply:
oh I see, although this solution works we should avoid modifying the array as we iterate over it, it also make the code hard to follow.
 You can define another array and add filtered results to it to keep responseData intact while iterating

Reply:
Ok got it :)  
I had considered using `.filter()` before but thought that `.splice()` is better in terms of memory.
Will update to use `.filter()` instead 👍